geomesa-arrow/geomesa-arrow-gt/src/main/scala/org/locationtech/geomesa/arrow/vector/ArrowAttributeReader.scala

```diff
@@ -174,56 +194,124 @@ object ArrowAttributeReader {
     */
   class ArrowDictionaryIntReader(accessor: NullableIntVector#Accessor,
                                  val dictionary: ArrowDictionary,
-                                 val dictionaryType: TypeBindings) extends ArrowAttributeReader with HasArrowDictionary {
+                                 val dictionaryType: TypeBindings) extends ArrowDictionaryReader {
+    private val holder = new NullableIntHolder
     override def apply(i: Int): AnyRef = {
-      if (accessor.isNull(i)) { null } else {
-        dictionary.lookup(accessor.get(i))
+      accessor.get(i, holder)
+      if (holder.isSet == 0) { null } else {
+        dictionary.lookup(holder.value)
+      }
+    }
+
+    override def getEncoded(i: Int): Int = accessor.get(i)
+  }
+
+  object ArrowGeometryReader {
+    def apply(vector: FieldVector, binding: Class[_], precision: EncodingPrecision): ArrowAttributeReader = {
+      if (binding == classOf[Point]) {
```

Comment:
yeah, ObjectType was originally written as a glue between our different serialization schemes. i think in that use case, the exact geometry type wasn't important - might be worth re-evaluating.